src/app.js

```diff
@@ -145,11 +145,13 @@ export default function(app) {
   function updateElementData(element, oldData, data) {
     for (var name in merge(oldData, data)) {
       var value = data[name]
-      var oldValue = oldData[name]
+      var oldValue = name === "value" || name === "checked"
```

Comment:
These are the two attributes whose actual value change when you interact with the component. So, when diffing an element's data, we want to look at the real DOM (not the stuff in oldData).